edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.MqttBrokerAdapter/brokerConnection/MqttBrokerConnector.cs

```diff
@@ -91,6 +94,13 @@ public async Task ConnectAsync(string serverAddress, int port)
 
             client.ConnectionClosed += this.TriggerReconnect;
```

Comment:
What happens if you do this before the first connect ([line 62](https://github.com/Azure/iotedge/pull/3369/files#diff-cfa187e87717a80d693ad56d57a9c703L62))? Does it try to connect right away?